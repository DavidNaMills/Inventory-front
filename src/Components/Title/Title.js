import React from 'react';
import classes from './title.module.scss';


const Title = (props) => {
    const { size = null, message = '', center = false } = props;
    let element = null;

    switch (size) {
        case 'h1':
            element = <h1 className={[classes.title_h1, center ? classes.title__center : null].join(' ')} >{message}</h1>
            break;
        case 'h2':
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
        case 'h3':
            element = <h3 className={[classes.title_h3, center ? classes.title__center : null].join(' ')}>{message}</h3>
            break;
        case 'h4':
            element = <h4 className={[classes.title_h4, center ? classes.title__center : null].join(' ')}>{message}</h4>
            break;
        case 'h5':
            element = <h5 className={[classes.title_h5, center ? classes.title__center : null].join(' ')}>{message}</h5>
            break;
        case 'h6':
            element = <h6 className={[classes.title_h6, center ? classes.title__center : null].join(' ')}>{message}</h6>
            break;
        default:
            element = <h2 className={[classes.title_h2, center ? classes.title__center : null].join(' ')}>{message}</h2>
            break;
    };

    return (
        <div>
            {element}
        </div>
    )
}

export default React.memo(Title);