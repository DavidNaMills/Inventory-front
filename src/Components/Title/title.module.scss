@import '../../styles/variables/colors.module.scss';
@import '../../styles/fonts/fontSizes.module.scss';
@import '../../styles/mixins/title.module.scss';

.title {
    @include setTitle($h2, $title-color);
    
    &_h1{
        @include setTitle($h1, $title-color);
    };

    &_h2{
        @include setTitle($h2, $title-color);
    };
    
    &_h3{
        @include setTitle($h3, $title-color);
    }

    &_h4{
        @include setTitle($h4, $title-color);
    }

    &_h5{
        @include setTitle($h5, $title-color);
    }

    &_h6{
        @include setTitle($h6, $title-color);
    }

    &__center{
        text-align: center;
    }


}