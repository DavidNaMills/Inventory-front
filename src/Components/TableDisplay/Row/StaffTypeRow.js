import React from 'react';
import { isTablet, withOrientationChange } from 'react-device-detect';
import Title from '../../Title/Title';
import classes from './ProductTypeRow/ProductTypeRow.module.scss';
import { useLocale } from '../../../Context/LocaleContext';

const StaffTypeRow = (props) => {
    const locale = useLocale();
    const { config, data, withTitle = true, showInfo=true } = props;

    const isRole = (key, txt) =>{
        if(key === 'role'){
            return locale[txt];
        } else {
            return txt;
        }
    }

    return (
        <div className={classes.productTypeRow}>
            {withTitle &&
                <div className={classes.productTypeRow__title}>
                    <Title message={data[config.main[0]]} size='h3' />
                </div>
            }
            {showInfo &&
                <div className={(isTablet && props.isPortrait) ? classes.productTypeRow__cellContPortrait : classes.productTypeRow__cellCont}>
                    <p><span>{locale[config.main[1]]}:</span>{isRole(config.main[1], data[config.main[1]])}</p>
                    <p><span>{locale[config.main[2]]}:</span>{isRole(config.main[2], data[config.main[2]])}</p>
                </div>
            }
        </div>
    )
}

export default withOrientationChange(StaffTypeRow);