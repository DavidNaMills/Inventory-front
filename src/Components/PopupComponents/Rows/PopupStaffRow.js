import React from 'react';
import { isMobileOnly, isMobile } from 'react-device-detect';
import StaffTypeRow from '../../TableDisplay/Row/StaffTypeRow';
import classes from '../Popup.module.scss';
import Title from '../../Title/Title';
import {useLocale} from '../../../Context/LocaleContext';

const config = {main: ['', 'name', 'role']};

const PopupStaffRow = (props) => {
    const { staff, isPortrait } = props;
    const locale = useLocale();

    const tempScrollableStaff = {
        overflowY: 'auto',
        height: isMobile ? isPortrait ? '20vh' : '10vw' : '15vh',
    }
    
    return (
        staff.length > 0 ? (
            <div className={classes.popup__color1}>
                <div className={classes.popup__mediumSpacing}>
                    <Title message={staff.length>0 ? `${locale['staff_count']}: ${staff.length} ${locale['in_ttl']}` : `${locale['no_staff_type']}`} size='h4' />
                </div>
                <div style={tempScrollableStaff}>
                    {
                        staff.map((c, d) =>
                            <div key={d} className={d % 2 ? classes.popup__row_odd : classes.popup__row_even}>
                                <StaffTypeRow
                                    config={config}
                                    data={c}
                                    withTitle={false}
                                    isMobileOnly={isMobileOnly}
                                />
                            </div>
                        )
                    }
                </div>
            </div>
        )
            : <p className={classes.popup__mediumSpacing}>No staff at this location</p>
    )
}

export default PopupStaffRow;