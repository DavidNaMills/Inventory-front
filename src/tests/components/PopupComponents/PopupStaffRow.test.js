import React from 'react';
import '../../testHelpers/context';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import PopupStaffRow from '../../../Components/PopupComponents/Rows/PopupStaffRow';
import StaffTypeRow from '../../../Components/TableDisplay/Row/StaffTypeRow';
import Title from '../../../Components/Title/Title';

const config = {main: ['', 'name', 'role']};
// const tempScrollableStaff = {
//     overflowY: 'auto',
//     maxHeight: isPortrait ? '30vh' : '15vw',
// }

const staff = [
    { name: 'david', role: '2' },
    { name: 'alan', role: '1' },
]


describe('<PopupStaffRow/> test suite', ()=>{
    let wrapper;

    beforeEach(()=>{
        wrapper = shallow(<PopupStaffRow staff={staff} isPortrait={false}/>);
    });

    it('matches snapshot', ()=>{
        const comp = renderer.create(<PopupStaffRow staff={staff} isPortrait={false}/>)
        const app = comp.toJSON();
        expect(app).toMatchSnapshot();
    });
    
    it('exists', ()=>{
        expect(wrapper.exists).toBeTruthy();
    });
    
    it('renders a <p> element with text if staff array prop is empty', ()=>{
        const tW = shallow(<PopupStaffRow staff={[]} isPortrait={false}/>);
        const p = tW.find('p');
        expect(p.length).toBe(1);
        expect(p.props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(p.text()).toEqual('No staff at this location');
    });
    
    it('renders a <div> element with className "popup__color1"', ()=>{
        const div = wrapper.find('div');
        expect(div.length).toBe(5);
        expect(div.at(0).props()).toHaveProperty('className', 'popup__color1');
    });
    
    it('renders a <Title/> component', ()=>{
        const div = wrapper.find('div');
        const ttl = div.at(1).find(Title);
        expect(div.at(1).props()).toHaveProperty('className', 'popup__mediumSpacing');
        expect(ttl.length).toBe(1);
        expect(ttl.props()).toHaveProperty('message', `Staff: ${staff.length} in total`);
        expect(ttl.props()).toHaveProperty('size', 'h4');
    });
    
    it('renders a <div> with styling if isPortrait = false', ()=>{
        const div = wrapper.find('div');
        expect(div.at(2).props().style).toHaveProperty('overflowY', 'auto');
        expect(div.at(2).props().style).toHaveProperty('maxHeight', '15vw');
    });
    
    it('renders a <div> with styling if isPortrait = true', ()=>{
        const tW = shallow(<PopupStaffRow staff={staff} isPortrait={true}/>);
        const div = tW.find('div');
        expect(div.at(2).props().style).toHaveProperty('overflowY', 'auto');
        expect(div.at(2).props().style).toHaveProperty('maxHeight', '30vh');
    });

    it('renders 2 <StaffTypeRow/> components', ()=>{
        const row = wrapper.find('div').at(2).find(StaffTypeRow);
        expect(row.length).toBe(2);
    });
});