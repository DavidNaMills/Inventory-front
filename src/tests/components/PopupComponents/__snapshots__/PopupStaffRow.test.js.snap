// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupStaffRow/> test suite matches snapshot 1`] = `
<div
  className="popup__color1"
>
  <div
    className="popup__mediumSpacing"
  >
    <div>
      <h4
        className="title_h4 "
      >
        No staff at this location: 2 in total
      </h4>
    </div>
  </div>
  <div
    style={
      Object {
        "maxHeight": "15vw",
        "overflowY": "auto",
      }
    }
  >
    <div
      className="popup__row_even"
    >
      <div
        className="productTypeRow"
      >
        <div
          className="productTypeRow__cellCont"
        >
          <p>
            <span>
              name
              :
            </span>
            david
          </p>
          <p>
            <span>
              role
              :
            </span>
            Sub-Admin
          </p>
        </div>
      </div>
    </div>
    <div
      className="popup__row_odd"
    >
      <div
        className="productTypeRow"
      >
        <div
          className="productTypeRow__cellCont"
        >
          <p>
            <span>
              name
              :
            </span>
            alan
          </p>
          <p>
            <span>
              role
              :
            </span>
            Staff
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
